userspace/falco/stats_writer.cpp

```diff
@@ -309,6 +309,73 @@ stats_writer::collector::collector(const std::shared_ptr<stats_writer>& writer)
 {
 }
 
+std::string merge_strings(const std::vector<std::string>& v, const std::string& delim) {
+	if(v.empty())
+	{
+		return "";
+	}
+	std::stringstream ss;
+	std::copy(v.cbegin(), v.cend(), std::ostream_iterator<std::string>(ss, delim.c_str()));
+	const std::string s_str = ss.str();
+	return s_str.substr(0, s_str.size() - delim.size());
+}
+
+void add_netinfo_metrics_output_fields(
+	nlohmann::json& output_fields,
+	const std::shared_ptr<sinsp>& inspector)
+{
+	const auto ipv4_ifinfo = inspector->get_ifaddr_list().get_ipv4_list();
+	const auto ipv6_ifinfo = inspector->get_ifaddr_list().get_ipv6_list();
+
+	// For each interface name, collect the corresponding list of IPv4/IPv6 addresses
+	std::map<std::string, std::vector<std::string>> ifnames_to_ipv4_addresses;
+	std::map<std::string, std::vector<std::string>> ifnames_to_ipv6_addresses;
+
+	std::for_each(ipv4_ifinfo->cbegin(), ipv4_ifinfo->cend(), [&ifnames_to_ipv4_addresses](const auto& ifinfo) {
```

Comment:
I appreciate the "modern C++"ism but in the codebase we normally use regular for loops. If we have enough reasons to switch to algorithms we could do it throughout the codebase but I'd rather stick with what we use everywhere :)

Reply:
Yes sure, no problem :)